libraries/default.rb

```diff
@@ -529,6 +529,15 @@ def configure_nginx_cloudwatch_logs
       configure_cloudwatch_log("nginx-access", "/var/log/nginx/access.log", "%d/%b/%Y:%H:%M:%S %z")
       configure_cloudwatch_log("nginx-error", "/var/log/nginx/error.log", "%d/%b/%Y:%H:%M:%S %z")
     end
+
+    def get_nginx_worker_procs
+      number_of_cpus = execute_command(%Q(nproc)).chomp.to_i
+      if admin_node? || engage_node?
+        [(number_of_cpus / 2), 4].max
```

Comment:
prod engage and admin have 32 cpus, so this would be 16. dev clusters would get a minimum of 4 (which is the current default setting)